src/resdk/resources/data.py

```diff
@@ -353,7 +357,21 @@ def download(self, file_name=None, field_name=None, download_dir=None):
             "{}/{}".format(self.id, fname)
             for fname in self.files(file_name, field_name)
         ]
-        self.resolwe._download_files(files, download_dir)
+
+        # Only applies if downloading a single file
+        custom_file_names = None
+        if custom_file_name:
+            if file_name or field_name:
+                custom_file_names = [custom_file_name] * len(files)
+            else:
+                logging.warning(
```

Comment:
I'd raise error, instead of logging a warning. In the case you provide custom file name, but do not specify the file_name / field_name (and there are multiple files on Data), errors will be raised anyway, but might not be as meaningful.